mslib/msui/qt5/ui_performance_dockwidget.py

```diff
@@ -0,0 +1,88 @@
+        PerformanceDockWidget.resize(767, 123)
+        self.gridLayout = QtWidgets.QGridLayout(PerformanceDockWidget)
+        self.gridLayout.setContentsMargins(9, 9, 9, 9)
+        self.gridLayout.setVerticalSpacing(1)
+        self.gridLayout.setObjectName("gridLayout")
+        self.horizontalLayout_2 = QtWidgets.QHBoxLayout()
+        self.horizontalLayout_2.setContentsMargins(-1, 8, -1, -1)
+        self.horizontalLayout_2.setObjectName("horizontalLayout_2")
+        self.label_2 = QtWidgets.QLabel(PerformanceDockWidget)
+        self.label_2.setIndent(5)
+        self.label_2.setObjectName("label_2")
+        self.horizontalLayout_2.addWidget(self.label_2)
+        self.lbAircraftName = QtWidgets.QLabel(PerformanceDockWidget)
+        self.lbAircraftName.setIndent(3)
+        self.lbAircraftName.setObjectName("lbAircraftName")
+        self.horizontalLayout_2.addWidget(self.lbAircraftName, 0, QtCore.Qt.AlignHCenter)
+        self.pbLoadPerformance = QtWidgets.QPushButton(PerformanceDockWidget)
+        self.pbLoadPerformance.setObjectName("pbLoadPerformance")
+        self.horizontalLayout_2.addWidget(self.pbLoadPerformance, 0, QtCore.Qt.AlignRight)
+        self.gridLayout.addLayout(self.horizontalLayout_2, 1, 1, 1, 1)
+        self.horizontalLayout_4 = QtWidgets.QHBoxLayout()
+        self.horizontalLayout_4.setContentsMargins(-1, -1, 0, -1)
+        self.horizontalLayout_4.setObjectName("horizontalLayout_4")
+        self.label_4 = QtWidgets.QLabel(PerformanceDockWidget)
+        self.label_4.setIndent(1)
+        self.label_4.setObjectName("label_4")
+        self.horizontalLayout_4.addWidget(self.label_4)
+        self.dsbFuel = QtWidgets.QDoubleSpinBox(PerformanceDockWidget)
+        self.dsbFuel.setObjectName("dsbFuel")
+        self.horizontalLayout_4.addWidget(self.dsbFuel)
+        self.gridLayout.addLayout(self.horizontalLayout_4, 3, 0, 1, 1)
+        self.horizontalLayout_3 = QtWidgets.QHBoxLayout()
+        self.horizontalLayout_3.setObjectName("horizontalLayout_3")
+        self.label_3 = QtWidgets.QLabel(PerformanceDockWidget)
+        self.label_3.setIndent(1)
+        self.label_3.setObjectName("label_3")
+        self.horizontalLayout_3.addWidget(self.label_3)
+        self.dteTakeoffTime = QtWidgets.QDateTimeEdit(PerformanceDockWidget)
+        self.dteTakeoffTime.setObjectName("dteTakeoffTime")
+        self.horizontalLayout_3.addWidget(self.dteTakeoffTime)
+        self.gridLayout.addLayout(self.horizontalLayout_3, 0, 0, 1, 1)
+        self.horizontalLayout_5 = QtWidgets.QHBoxLayout()
+        self.horizontalLayout_5.setContentsMargins(-1, 4, -1, -1)
+        self.horizontalLayout_5.setObjectName("horizontalLayout_5")
+        self.label_5 = QtWidgets.QLabel(PerformanceDockWidget)
+        self.label_5.setIndent(1)
+        self.label_5.setObjectName("label_5")
+        self.horizontalLayout_5.addWidget(self.label_5)
+        self.dsbTakeoffWeight = QtWidgets.QDoubleSpinBox(PerformanceDockWidget)
```

Comment:
Modifying this does not update the Table.

Reply:
Changing the values of the spinbox actually changes the values in the waypoints model. Even though the table is triggered with a dataChanged.emit, the table doesn't seem to update it in the table, only for the spinboxes. Same goes for the other spinbox. It seems that i'm stuck at this point. could please provide some assistance @ReimarBauer ?

Reply:
I will try to have a look at this.

Reply:
OK, this here seems to fix the issue:
```
 (e) base  ~  src  mss  git diff                                                                                                                 table-view-issue-200 
diff --git a/mslib/msui/tableview.py b/mslib/msui/tableview.py
index 9293cd84..3745bbb8 100644
--- a/mslib/msui/tableview.py
+++ b/mslib/msui/tableview.py
@@ -94,6 +94,7 @@ class MSSTableViewWindow(MSSViewWindow, ui.Ui_TableViewWindow):
         self.waypoints_model.update_distances(0)
         self.waypoints_model.save_settings()
         self.resizeColumns()
+        self.tableWayPoints.viewport().repaint()
 
     def openTool(self, index):
         """Slot that handles requests to open tool windows.
```
I got the solution from here
https://stackoverflow.com/questions/20256493/qtableview-doesnt-respond-to-datachanged-when-not-focused
From that link, I suspect the issue occurs more often than just. I would suggest to connect the dataChanged signal with a viewport-update-function-slot as suggested. A bit more googling might be insightful.

Reply:
Okay, I'll look into it.

Reply:
I have fixed the update issue and also added required test for the new dock widget.